### PR TITLE
aded tf for doing route53 for cert-manager DNS01 challenges

### PR DESCRIPTION
**Pull request by sidler-mozilla**

### CODE REVIEW

**Review by limed — COMMENTED** [review]
Once you created the role over you can annotate the service account as stated in my review here https://github.com/mozilla-it/helm-charts/pull/73#pullrequestreview-514287580

Example

```hcl
locals {
  cert_manager_settings = {
    "serviceAccount.annotations.eks\\.amazonaws\\.com/role-arn" = module.cert_manager_role.this_iam-role_arn
  }
}
```

**Review by limed — CHANGES_REQUESTED** [review]
See review https://github.com/mozilla-it/itse-apps-stage-1-infra/pull/45#pullrequestreview-514290030

File: terraform/cert-manager.tf [hunk 1]
```diff
@@ -13,6 +20,57 @@ resource "kubernetes_namespace" "cert_manager" {
...
 
+data "aws_iam_policy_document" "cert_manager" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "route53:ChangeResourceRecordSets",
+      "route53:ListResourceRecordSets"
+    ]
+    resources = [
+      "arn:aws:route53:::hostedzone/Z06469063HC60S7SHF706"
```
[the remark this answers is not in the document]
Comment (sidler-mozilla) [reply terraform/cert-manager.tf:31]: great suggestion!

File: terraform/cert-manager.tf [hunk 2]
```diff
@@ -28,7 +32,7 @@ data "aws_iam_policy_document" "cert_manager" {
       "route53:ListResourceRecordSets"
     ]
     resources = [
-      "arn:aws:route53:::hostedzone/Z06469063HC60S7SHF706"
+      "${data.aws_route53_zone.this.id}"
```
Comment (limed) [inline terraform/cert-manager.tf:35]: This won't work `id` just returns `Z06469063HC60S7SHF706` you need to do `"arn:aws:route53:::hostedzone/${data.aws_route53_zone.this.id}"`

**Review by limed — CHANGES_REQUESTED** [review]

File: terraform/cert-manager.tf (hunk 2 above)
Comment (sidler-mozilla) [reply terraform/cert-manager.tf:35]: ok

**Review by limed — APPROVED** [review]
r+

**Pull request merged**